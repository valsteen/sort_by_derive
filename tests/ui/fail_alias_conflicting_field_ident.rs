use sort_by_derive::EnumAccessor;

#[derive(EnumAccessor)]
#[accessor(b: u16, except(Variant3))]
#[accessor(a as b: u32, except(Variant1,Variant2))]
enum E {
    Variant1 { b: u16 },
    Variant2 { b: u16, c: u32 },
    Variant3 { a: u32 },
}

fn main() {}
