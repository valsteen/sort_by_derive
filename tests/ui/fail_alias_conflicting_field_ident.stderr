error: Duplicate accessor b
 --> tests/ui/fail_alias_conflicting_field_ident.rs:5:12
  |
5 | #[accessor(a as b: u32, except(Variant1,Variant2))]
  |            ^^^^^^

error: b is previously defined here
 --> tests/ui/fail_alias_conflicting_field_ident.rs:4:12
  |
4 | #[accessor(b: u16, except(Variant3))]
  |            ^^^^^^
